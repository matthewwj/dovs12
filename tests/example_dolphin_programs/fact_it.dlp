void printInt(i : int){
   output_string(int_to_string(i),get_stdout());
   output_string("\n",get_stdout());
}

int fact(n: int) {
    var x = n;
    var res = 1;
    while (x > 1) {
        res = res * x;
        x = x-1;
    };
    return res;
}

int main() {
    printInt(fact(5)); 
    printInt(fact(11)); 
    return 0;
}