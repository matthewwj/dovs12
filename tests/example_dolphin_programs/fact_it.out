120
39916800
0
